### ISSUE TITLE
[Bug] VERIFICATION_ERROR when publishing code

### ISSUE THREAD

**Issue by gregnazario** [aptos-labs/aptos-core#11595]
# 🐛 Bug

To report a security issue, please email security@aptoslabs.com.

## To reproduce

**Code snippet to reproduce**
(Repeatable on 2.2.2, 2.3.2, and 2.4.0) local / devnet / testnet
[vm_bug.zip](https://github.com/aptos-labs/aptos-core/files/13865293/vm_bug.zip)

```bash
aptos move publish --profile local --named-addresses vmbug_addr=local
Compiling, may take a little while to download git dependencies...
UPDATING GIT DEPENDENCY https://github.com/aptos-labs/aptos-core.git
UPDATING GIT DEPENDENCY https://github.com/aptos-labs/aptos-core.git
INCLUDING DEPENDENCY AptosFramework
INCLUDING DEPENDENCY AptosStdlib
INCLUDING DEPENDENCY AptosToken
INCLUDING DEPENDENCY MoveStdlib
BUILDING vm_bug
package size 3188 bytes
{
  "Error": "Simulation failed with status: Transaction Executed and Committed with Error VERIFICATION_ERROR\nExecution failed with status: CORE DUMP: >>>>>>>>>>>>\nCall stack:\n*frame #0: 0xb98022721f97da65950435386a5209be2fc54bed5f12fc8ea0a1c3d0c76f562f::vmbug::init_module [pc = 28]:\n0> VecPack(3, 0)\n1> StLoc(2)\n2> CopyLoc(0)\n3> Call(2)\n4> StLoc(1)\n5> MutBorrowLoc(2)\n6> MoveLoc(1)\n7> VecPushBack(3)\n8> MoveLoc(0)\n9> CallGeneric(0)\n10> CopyLoc(2)\n11> MoveLoc(2)\n12> LdU32(1)\n13> LdU64(1)\n14> LdU64(1000)\n15> CallGeneric(1)\n16> CallGeneric(2)\n17> CallGeneric(3)\n18> LdU256(1)\n19> LdU256(1)\n20> CallGeneric(4)\n21> CallGeneric(5)\n22> CallGeneric(6)\n23> CallGeneric(7)\n24> CallGeneric(7)\n25> LdU64(0)\n26> LdU256(0)\n27> Pack(4)\n28* MoveTo(StructDefinitionIndex(4))\nLocals (1633787a8):\n[0] Invalid\n[1] Invalid\n[2] Invalid\nOperand Stack:\n\n<<<<<<<<<<<<\n"
}
```

**Stack trace/error message**
```
Simulation failed with status: Transaction Executed and Committed with Error VERIFICATION_ERROR
Execution failed with status: CORE DUMP: >>>>>>>>>>>>
Call stack:
*frame #0: 0xb98022721f97da65950435386a5209be2fc54bed5f12fc8ea0a1c3d0c76f562f::vmbug::init_module [pc = 28]:
0> VecPack(3, 0)
1> StLoc(2)
2> CopyLoc(0)\n3> Call(2)
4> StLoc(1)
5> MutBorrowLoc(2)
6> MoveLoc(1)
7> VecPushBack(3)
8> MoveLoc(0)
9> CallGeneric(0)
10> CopyLoc(2)
11> MoveLoc(2)
12> LdU32(1)
13> LdU64(1)
14> LdU64(1000)
15> CallGeneric(1)
16> CallGeneric(2)
17> CallGeneric(3)
18> LdU256(1)
19> LdU256(1)
20> CallGeneric(4)
21> CallGeneric(5)
22> CallGeneric(6)
23> CallGeneric(7)
24> CallGeneric(7)
25> LdU64(0)
26> LdU256(0)
27> Pack(4)
28* MoveTo(StructDefinitionIndex(4))
Locals (1633787a8):
[0] Invalid
[1] Invalid
[2] Invalid
Operand Stack:

<<<<<<<<<<<<
```

## Expected Behavior

No coredump

## System information

**Please complete the following information:**
- Aptos Core Version
- Rust Version
- Computer OS

**Comment by gregnazario** [comment]
This was repeatable on various older versions of the local testnet, so it's been around a while.

**Comment by georgemitenkov** [comment]
Fixed by #11656. The issue is not a bug, but bad naming and error propagation. The error was caused by exceeding the type size limit inside interpreter loop.
